### DISCUSSION TITLE
waveshare rs485 to eth wiring doubts

### DISCUSSION THREAD

**Discussion by horamarques** [wills106/homeassistant-solax-modbus#120] [General]
How do you power the waveshare rs485 2 eth? 

I couldn't find in the manual the type of transformer to use, only says 9 to 12v DC...

Thanks

**Comment by wills106** [comment]
You might get a DC PSU with it?
You can get DIN Rail mounted 12v DC PSU's which might be more convenient?

**Reply by infradom** [reply]
my waveshare accepts 9-24VDC , so it is very flexible when in comes to DC power

**Reply by horamarques** [reply]
I'm gone use as power supply: DIN 24VDC 0.63A 15W - PCEnergy PCE15-24-0,63-DIN-S
hope it has enough power...

**Reply by infradom** [reply]
I am almost sure the Waveshare device does not consume more than 3 watt. If it would consume more, I would feel warm if I touch it

**Reply by arwooldridge** [reply]
Solax Hybrid Gen4 inverters output a suitable voltage +13v and Gnd on the com port pins 3 and 6, I use it to drive my waveshare rs485 to eth.
PS double check the documentation to be sure!
